### ISSUE TITLE
Attempted to retain deallocated object

### ISSUE THREAD

**Issue by MoralesJohn** [AssistoLab/DropDown#108]
So, I have a project that already includes a few DropDown menus in one view controller. I am trying to add a drop down to a new controller, and getting the above error. I did read issue #20 [https://github.com/AssistoLab/DropDown/issues/20](url), but I am already using the proposed solution. 

My code: 

 ```
import DropDown

 class DashboardViewController: UIViewController, UIScrollViewDelegate {
    @IBOutlet weak var shareButton : UIButton!
    let shareMenu = DropDown()
    
    @IBAction func shareButtonPressed(_ sender: Any) {
        shareMenu.show()
    }

...

    // called from viewDidLoad
    func shareMenuSetup() {
        shareMenu.anchorView = shareButton
        let shareOptions = ["Past Week", "Past Month", "History to Date"]
        shareMenu.dataSource = shareOptions
        shareMenu.selectionAction = { [unowned self] (index: Int, item: String) in
            //stuff
        }
    }
```

All of which is essentially what I have in the view controller where I am using DropDown successfully. I can comment out everything else, and as soon as I add the line `let shareMenu = DropDown()`, I get the deallocation error. 

Any suggestions?

**Comment by MoralesJohn** [comment]
Per issue #47, I changed my declaration to `var shareMenu = DropDown?` and added `shareMenu = DropDown()` to the viewDidLoad, and now it is working. Not sure why in needs to be an optional in one view controller and not in the other, though.

**Comment by kevin-hirsch** [comment]
Hello!
I've pushed a new commit that should fix this problem (80a125b on master)
